sigstore/sign.py

```diff
@@ -175,9 +174,17 @@ def _signing_cert(
     def sign(
         self,
         input_: IO[bytes],
```

Comment:
Note: I considered using `input_ IO[bytes] | bytes` where `bytes` type would be used with `Prehashed` but I think it further complicates the API. So I've kept `IO[bytes]` instead. Please advise.